subprojects/logging/src/main/java/org/gradle/internal/logging/slf4j/WarningRewriterBuildOperationAwareLogger.java

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.slf4j;
+
+import org.gradle.api.logging.LogLevel;
+import org.gradle.api.logging.Logger;
+import org.gradle.internal.Cast;
+import org.gradle.internal.operations.OperationIdentifier;
+import org.slf4j.Marker;
+
+import javax.annotation.Nullable;
+
+/**
+ * A {@link BuildOperationAwareLogger} that can rewrite or silence warnings.
+ */
+public class WarningRewriterBuildOperationAwareLogger extends BuildOperationAwareLogger {
```

Comment:
Not really because both implementations end up being build operation related.
But the rewriting hook could be pulled up into `DefaultContextAwareTaskLogger` simplifying the change, will try that.